trunner/target/base.py

```diff
@@ -11,25 +11,42 @@
 from trunner.types import TestOptions, TestResult
 
 
-def find_port(port_hint: str) -> str:
-    port = None
-
-    for p in list_ports.grep(port_hint):
```

Comment:
After some thinking, we can simply pass a part of port name, which is correct for both older and newer board. For older the device string from `list_ports()` is:
```
/dev/ttyACM0 - DAPLink CMSIS-DAP - mbed Serial Port
```

for newer one:

```
/dev/ttyACM0 - MCU-LINK on-board (r0E2) CMSIS-DAP V2.250 - VCOM
```

So I propose looking for `CMSIS-DAP` for armv7m7 targets. Then all this additional code is not needed.